### ISSUE TITLE
Include a performance comparison in README

### ISSUE THREAD

**Issue by dtran320** [tommikaikkonen/redux-orm#1]
This most applies to https://github.com/tommikaikkonen/redux-orm/tree/feature/related:

It would be useful to see a performance comparison for a long series of actions/reducers making changes to the state (in a similar format to tests at http://jsperf.com) and publish this as part of the README for this project.

**Comment by tommikaikkonen** [comment]
I agree, the library should be performance tested, particularly with larger datasets. It's possible to implement optimizations in `Backend`, but that'll add to the build size and complexity. Performance tests should indicate if that's a worthwhile tradeoff.

Keeping this open.

**Comment by dagatsoin** [comment]
+1 I am actually comparing redux + redux-orm VS mobx.js
Both approaches are neat but the "warn" about the performance compared to bare Redux is a blocking element for a comparaison.

**Comment by tommikaikkonen** [comment]
Whether the performance is an issue depends on the type of data you're handling. In our production we have 5-100 entities for each type and have not noticed issues.

Write performance has a larger penalty than reads at the moment, because redux-orm usually needs to do multiple internal passes for each C/U/D operation. Since they work immutably they can end up creating a lot of new objects that end up discarded. I've started work on batching the internal passes -- I wrote a lightweight utility library that has an equivalent API for mutable and immutable operations as the first building block. https://github.com/tommikaikkonen/immutable-ops

**Comment by tommikaikkonen** [comment]
No performance comparison yet, but for anyone reading this issue, the new 0.8.1 release uses batched mutations. This makes working with large updates and big collections considerably faster.

**Comment by dillionverma** [comment]
Being new to this library and having extensive rails experience, I am very intrigued by the syntactic similarity, but I am holding back because of doubts about performance for a large app. I am sure others have had similar thoughts and would like to see a quantitative comparison for a real-world example before using it for themselves.

Hoping to resurrect this issue.

**Comment by haveyaseen** [comment]
@dillionverma With v0.10.0 we now introduced memoization by instance, i.e. we try to prevent full table scans when possible. Further, rows are now quickly retrieved if an `id` is provided independent on the order of QuerySet clauses/transformations. And many-to-many relations have been optimized using ES6 `Set`s.

I think it's not really feasible to make a comparison based on a single large app as the differences between two large applications will be way too significant. But each test now outputs performance statistics of certain operations as well (which also must not exceed predefined thresholds).

Otherwise, `redux-orm`'s performance can't really be measured in a single digit. It depends to a large degree on the application itself and what it really does. If mutable sessions are used, there is probably little in the JavaScript world that is faster than `redux-orm`. If immutable sessions have to be used, `immutable-ops`' batched mutations combined with selectors are quite performant, too. But then again it really depends on what you're doing with the data stored in the ORM.
